Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -325,6 +332,24 @@ private extension HTTPClient {
 
 // MARK: - Extensions
 
+private extension HTTPRequest {
+
+    func headers(with authHeaders: HTTPClient.RequestHeaders) -> HTTPClient.RequestHeaders {
+        var result: HTTPClient.RequestHeaders = [:]
+
+        if self.path.authenticated {
+            result += authHeaders
+        }
+
+        if let nonce = self.nonce {
+            result += HTTPClient.nonceHeader(with: nonce)
+        }
```

Comment:
is there any situation where the nonce should be `nil` but the `path.authenticated = true`? 
If not, we should try to consolidate the conditions and probably log if they're not simultaneously met

Reply:
The `/health` endpoint is one example.